src/commands/txCommands/pullAll.js

```diff
@@ -0,0 +1,90 @@
+const chalk = require('chalk');
+const child_process = require('child_process');
+const Rx = require('rxjs');
+const txlib = require('./util');
+const pullResourceTrns = require('./util/pullResourceTrns');
+
+const getProjectResourcesList$ =
+	txlib.resources$
+		.flatMap(Rx.Observable.from);
+
+const child_process$ = Rx.Observable.bindNodeCallback(child_process.exec);
+
+/**
+ * Creates git commit of what has been staged
+ *
+ * note: adjusting resource name in git commit message to prevent jira ticket
+ * from moving through workflow. We have a script which looks at all git commits
+ * to see if a ticket number is contained within it and if so associates a branch/PR
+ * with it. We don't want that to happen with tx pulls.
+ * @param  {String} resource resource slug to be used when we pull translations
+ * @param  {Observable} observer Observable of resource pull
+ * @return {Observable} Observable of shell process
+ */
+const commitResource = (resource, observer) =>
+	child_process$(`git commit -m "tx:pull for ${resource.replace(/-/g, '_')}" --no-verify`)
+		.flatMap(() => {
+			console.log(chalk.green(`- commited changes for '${resource}'`));
+			// start next tx:pull
+			return observer.complete();
+		});
+
+/**
+ * Called when a tx pull is complete for a particular resource
+ * @param  {String} resource resource slug to be used when we pull translations
+ * @param  {Observable} observer Observable of resource pull
+ * @param  {Boolean} commitOnComplete flag used to determine if we should commit the resource in git
+ * @return {undefined}
+ */
+const onPullResourceComplete = (resource, observer, commitOnComplete) => {
+	console.log(chalk.green('done...', resource));
+	if (!commitOnComplete) {
+		console.log('here...');
+		return observer.complete();
+	}
+
+	// stops here...
+	console.log('or here....');
+
+	child_process$('git status')
+		.flatMap((error, stdout, stderr) => {
+			console.log('stdout', stdout);
+			if (stdout.includes('modified:')) {
```

Comment:
So refactor still in progress, but running into a new issue where if i request to commit it never makes it to this point. As as if the observer stops waiting. Do you have any insights @mmcgahan?

Reply:
does it stop here or on line 46?

Assuming it correctly logs on line 46 and 51, check how `bindNodeCallback` works - it will _not_ emit the `error` argument (it will treat that as an error in the Observable instead), and will convert the separate `stdout, stderr` args into a single array of `[stdout, stderr]` in the thing that it emits